parcels/field.py

```diff
@@ -637,7 +646,14 @@ def from_netcdf(
 
     @classmethod
     def from_xarray(
-        cls, da, name, dimensions, mesh="spherical", allow_time_extrapolation=None, time_periodic=False, **kwargs
+        cls,
+        da: xr.DataArray,
+        name: str,
+        dimensions,
+        mesh: Mesh = "spherical",
+        allow_time_extrapolation: bool | None = None,
```

Comment:
It's because it is set as the default option in the signature, and somewhere in the code must deal with the None case. I agree that this can be refactored in a future PR

Reply:
Hmm, would now not be a good opportunity to change the default to `False` throughout. 

I dug a little through the code, and I think the default is None because of this logic. 
https://github.com/OceanParcels/Parcels/blob/e146ff344f734fd44061512bfbcb84d7c43967af/parcels/field.py#L211-L214
But there must be another way to implement this?

Reply:
In that case, I think it makes sense. Where its treated as a static field if `allow_time_extrapolation` is None and the time dims are one. We could instead make it explicit that they have to set it to `True` for static fields?

```python
if not allow_time_extrapolation and len(self.grid.time) == 1: 
     raise ValueError("Explicitly set `allow_time_extrapolation` to True when working with static fields.")

self.allow_time_extrapolation = allow_time_extrapolation
```

Reply:
I'm inclined to make this a separate issue, along with an issue about a Parcels deprecation policy

Reply:
OK, now I understand. Your proposed change would indeed have a major impact; not least on many, many unit tests with time-stationary fields. So I agree then it's not worth the change just for the slighter cleaner type-annotations

Reply:
Cool, lets keep the current behaviour then